test/controllers/AddressControllerSpec.scala

```diff
@@ -181,7 +181,7 @@ class AddressControllerSpec extends BaseSpec with MockitoSugar {
       when(mockAuthJourney.authWithPersonalDetails).thenReturn(new ActionBuilder[UserRequest] {
         override def invokeBlock[A](request: Request[A], block: UserRequest[A] => Future[Result]): Future[Result] =
           block(
-            UserRequest(Some(Fixtures.fakeNino), None, None, NonFilerSelfAssessmentUser, Credentials("", "GovernmentGateway"), ConfidenceLevel.L200, None, None, None, None, None, request)
+            UserRequest(Some(Fixtures.fakeNino), None, None, NonFilerSelfAssessmentUser, Credentials("", "GovernmentGateway"), ConfidenceLevel.L200, None, None, None, None, None, None, request)
```

Comment:
`UserRequests` are created a lot in the tests. The one small change you made has meant you have had to change 35 files most of which are just adding `None` to this object. Try creating something along the lines of the builder pattern for creating these in tests. Something like
```
case class UserRequestBuilder(private userRequest: UserRequest) {
 
  def withFirstName(firstName: String): UserRequestBuilder = ???
  def withLastName(lastName: String): UserRequestBuilder = ???
  ....
}

object UserRequestBuilder {

  def apply(nino: Nino) = {
    val userRequest =
        UserRequest(Some(nino), None, None, NonFilerSelfAssessmentUser, Credentials("", "GovernmentGateway"), ConfidenceLevel.L200, None, None, None, None, None, None, FakeRequest())
    UserRequestBuilder(userRequest)
  }

}
```
If you use `val quicklens = "com.softwaremill.quicklens" %% "quicklens" % "1.4.12"` this will greatly reduce the complexity of modifying deeply nested structures like name